File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.py
class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:

        # Find the max_candies in the array
        max_candies = 0
        for candy in candies:
            max_candies = max(max_candies, candy)

        result = []

        # Calcuate the boolena array by comparing the sum of current candy and extra_candies with the max_candies 
        for candy in candies:
            result.append(True if extraCandies + candy >= max_candies else False )

        return result

File: 1528-kids-with-the-greatest-number-of-candies/1528-kids-with-the-greatest-number-of-candies.java
class Solution {
    public List<Boolean> kidsWithCandies(int[] candies, int extraCandies) {
        // Find the max candies amonst the kids i.e. in the arrays
        int maxCandies = 0;

        for(int candy: candies){
            maxCandies = Math.max(candy, maxCandies);
        }

        // Prepare the array to be returned
        List<Boolean> answer = new ArrayList<>();

        // Calculate the array by adding the current candies with extra ones and comparing it with maxCandies
        for(int candy: candies){
            answer.add(maxCandies<=(candy+extraCandies)?true:false);
        }

        return answer;
    }
}